File: migrate_db.py
#!/usr/bin/env python3
"""
Database Migration: Add annotated_image_path column to cam2_recordings
"""

import pymysql
import sys

db_config = {
    'host': 'localhost',
    'database': 'wagodb',
    'user': 'gh',
    'password': 'a12345'
}

def main():
    try:
        print("Connecting to database...")
        conn = pymysql.connect(**db_config)
        cursor = conn.cursor()

        # Check if column exists
        cursor.execute('SHOW COLUMNS FROM cam2_recordings LIKE "annotated_image_path"')
        result = cursor.fetchone()

        if result:
            print('✓ Column annotated_image_path already exists')
        else:
            print('Adding column annotated_image_path...')
            cursor.execute('''
                ALTER TABLE cam2_recordings
                ADD COLUMN annotated_image_path VARCHAR(255) DEFAULT NULL
                AFTER analyzed
            ''')
            conn.commit()
            print('✓ Column added successfully')

        # Verify
        cursor.execute('DESCRIBE cam2_recordings')
        print('\nTable structure:')
        for row in cursor.fetchall():
            print(f"  {row[0]}: {row[1]}")

        cursor.close()
        conn.close()
        print('\n✓ Migration completed successfully')
        return 0

    except Exception as e:
        print(f'✗ Error: {e}', file=sys.stderr)
        return 1

if __name__ == '__main__':
    sys.exit(main())

File: migrate_db.php
#!/usr/bin/env php
<?php
/**
 * Database Migration: Add annotated_image_path column to cam2_recordings
 */

require_once __DIR__ . '/config.php';

try {
    echo "Connecting to database...\n";
    $pdo = getDbConnection();

    // Check if column exists
    $stmt = $pdo->query("SHOW COLUMNS FROM cam2_recordings LIKE 'annotated_image_path'");
    $result = $stmt->fetch();

    if ($result) {
        echo "✓ Column annotated_image_path already exists\n";
    } else {
        echo "Adding column annotated_image_path...\n";
        $pdo->exec("
            ALTER TABLE cam2_recordings
            ADD COLUMN annotated_image_path VARCHAR(255) DEFAULT NULL
            AFTER analyzed
        ");
        echo "✓ Column added successfully\n";
    }

    // Verify
    echo "\nTable structure:\n";
    $stmt = $pdo->query("DESCRIBE cam2_recordings");
    while ($row = $stmt->fetch(PDO::FETCH_ASSOC)) {
        echo "  {$row['Field']}: {$row['Type']}\n";
    }

    echo "\n✓ Migration completed successfully\n";

} catch (PDOException $e) {
    echo "✗ Error: " . $e->getMessage() . "\n";
    exit(1);
}
